pkg/services/librarypanels/database.go

```diff
@@ -110,6 +131,26 @@ func (lps *LibraryPanelService) disconnectDashboard(c *models.ReqContext, uid st
 	})
 }
 
+// disconnectLibraryPanelsForDashboard deletes connections for all Library Panels in a Dashboard.
+func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID int64, panelCount int64) error {
+	if panelCount == 0 {
+		return nil
+	}
+	return lps.SQLStore.WithTransactionalDbSession(context.Background(), func(session *sqlstore.DBSession) error {
+		result, err := session.Exec("DELETE FROM library_panel_dashboard WHERE dashboard_id=?", dashboardID)
+		if err != nil {
+			return err
+		}
+		if rowsAffected, err := result.RowsAffected(); err != nil {
+			return err
+		} else if rowsAffected != panelCount {
+			return fmt.Errorf("disconnected %d panels, while expecting %d", rowsAffected, panelCount)
```

Comment:
As discussed previously, I think it's a bad idea to return an error if a different number of connections were removed than expected. The reason is that a parallel process may have added or removed connections in the meantime. I think the only concern we should have here is whether connections were left behind in the database, and if they were it should only be logged as a warning, not returned as an error.

Reply:
Aah, ok now I understand what you mean. That makes sense.